scipy/sparse/_matrix_io.py

```diff
@@ -165,3 +165,111 @@ def load_npz(file):
         else:
             raise NotImplementedError(f'Load is not implemented for '
                                       f'sparse matrix of format {sparse_format}.')
+
+def from_binsparse(arr, /, *, device=None, copy: bool | None = None):
+    from scipy.sparse import csr_array, csc_array
+    desc = arr.__binsparse_descriptor__()
+    arrs = arr.__binsparse__()
+
+    desc = desc["binsparse"]
+    version_tuple: tuple[int, ...] = tuple(int(v) for v in desc["version"].split("."))
+    if version_tuple != (0, 1):
+        raise RuntimeError("Unsupported `__binsparse__` protocol version.")
+
+    format = desc["format"]
+    format_err_str = f"Unsupported format: `{format!r}`."
+
+    if isinstance(format, str):
```

Comment:
Is this check needed? I think `format` has to always be a string? Things go wrong here if it isn't.